azure_arc_servers_jumpstart/docs/gcp_terraform.md

```diff
@@ -0,0 +1,117 @@
+
+    <!-- ![](../img/gcp/05.png) -->
+
+* Then click Enable to enable API access.
+
+    <!-- ![](../img/gcp/01.png) -->
+
+* Next, set up a service account key, which Terraform will use to create and manage resources in your GCP project. Go to the [create service account key page](https://console.cloud.google.com/apis/credentials/serviceaccountkey). Select the default service account or create a new one, give it a name, select Project then Owner as the role, JSON as the key type, and click Create. This downloads a JSON file with all the credentials that will be needed for Terraform to manage the resources. Copy the downloaded JSON file to the ```azure_arc_servers_jumpstart/gcp/terraform``` directory.
+
+* Finally, make sure your SSH keys are available in ```~/.ssh``` and named id_rsa.pub and id_rsa. If you followed the ssh-keygen guide above to create your key then this should already be setup correctly. If not, you may need to modify ```main.tf``` to use a key with a different path.
+
+### Create Azure Service Principal (SP)   
+
+* To connect the GCP virtual machine to Azure Arc, an Azure Service Principal assigned with the "Contributor" role is required. To create it, login to your Azure account run the following command:
+
+    ```az login```
+
+    ```az ad sp create-for-rbac -n "http://AzureArcGCP" --role contributor```
+
+    Output should look similar to this:
+
+    ```
+    {
+    "appId": "aedXXXXXXXXXXXXXXXXXXac661",
+    "displayName": "AzureArcGCP",
+    "name": "http://AzureArcGCP",
+    "password": "b54XXXXXXXXXXXXXXXXXb2338e",
+    "tenant": "72f98XXXXXXXXXXXXXXXXX11db47"
+    }
+    ```
+
+    **Note**: It is optional but highly recommended to scope the SP to a specific [Azure subscription and Resource Group](https://docs.microsoft.com/en-us/cli/azure/ad/sp?view=azure-cli-latest)
+
+# Deployment
+
+Before executing the Terraform plan, you must export the environment variables which will be used by the plan. These variables are based on the Azure Service Principal you've just created, your Azure subscription and tenant, and the GCP project name.
+
+* Retrieve your Azure Subscription ID and tenant ID using the ```az account list``` command.
+
+* The Terraform plan creates resources in both Microsoft Azure and Google Cloud. It then executes a script on a Google Cloud virtual machine to install the Azure Arc agent and all necessary artifacts. This script requires certain information about your Google Cloud and Azure environments. Edit ***scripts/vars.sh*** and update each of the variables with the appropriate values.
+    ![](../img/gcp/06.png)
+
+* Open Az CLI and navigate to the ```azure_arc_servers_jumpstart/gcp/terraform``` directory of the cloned repo.
+
+* Export the environment variables you edited by running the vars.sh script with the source command as shown below.
+
+    ```source ./scripts/vars.sh```
+
```

Comment:
- Since it's not a command rather a location, change ```azure_arc_servers_jumpstart/gcp/terraform``` to *azure_arc_servers_jumpstart/gcp/terraform*
- Environment variables should be injected and not run by the user. What am I missing? I see that in the arc_agent script.

Reply:
discussed with you and resolved